File: UtrechtJUG/KotlinMigration/src/main/java/nl/blue4it/basic/service/async/KafkaClient.java
package nl.blue4it.basic.service.async;

import example.avro.Payment;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
@Slf4j
public class KafkaClient {
    private final KafkaTemplate<String, Payment> kafkaTemplate;

    public void send(Payment payment) {
        System.out.println("Kafka client started");
        System.out.println("Kafka client executed");
    }
}

File: VoxxedCluj/KotlinMigration/src/main/java/nl/blue4it/basic/service/async/KafkaClient.kt
package nl.blue4it.basic.service.async

import example.avro.Payment
import lombok.RequiredArgsConstructor
import lombok.extern.slf4j.Slf4j
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Component

@Component
class KafkaClient {
    private val kafkaTemplate: KafkaTemplate<String, Payment>? = null

    fun send(payment: Payment) {
        println("Kafka client started")
        println("Kafka client executed")
    }
}